tests/publiccloud/run_ltp.pm

```diff
@@ -121,15 +125,67 @@ sub cleanup {
     type_string('', terminate_with => 'ETX');
 
     upload_logs('ltp_log.raw', failok => 1);
-    parse_extra_log(LTP => "$root_dir/ltp_log.json") if (script_run("test -f $root_dir/ltp_log.json") == 0);
-
+    if (script_run("test -f $root_dir/ltp_log.json") == 0) {
+        my $known_issues = download_whitelist();
+        process_ltp_known_issues("$root_dir/ltp_log.json") if $known_issues;
+        parse_extra_log(LTP => "$root_dir/ltp_log.json");
+    }
     if ($self->{my_instance} && script_run("test -f $root_dir/log_instance.sh") == 0) {
         assert_script_run($root_dir . '/log_instance.sh stop ' . $self->instance_log_args());
         assert_script_run("(cd /tmp/log_instance && tar -zcf $root_dir/instance_log.tar.gz *)");
         upload_logs("$root_dir/instance_log.tar.gz", failok => 1);
     }
 }
 
+sub process_ltp_known_issues {
+    my ($self, $ltp_log) = @_;
+    my $decoded_ltp_log = Mojo::JSON::from_json($ltp_log);
+    my $env             = gen_ltp_env();
+    my $suite           = get_required_var('COMMAND_FILE');
+
+    foreach my $test (@{$decoded_ltp_log->{results}}) {
+        my @issues;
+        if (($test->{'status'} eq 'fail' || $test->{'status'} eq 'brok' || $test->{'status'} eq 'warn') && ref($known_issues->{$suite}) eq 'ARRAY') {
+            my $entry = find_whitelist_entry($env, $suite, $test->{test_fqn});
+            bmwqemu::diag(sprintf("Failure in LTP:%s:%s is known, overriding to softfail", $suite, $test->{test_fqn}));
+            $test->{status} = 'softfail';
+        }
+    }
+    my $final_ltp_log = Mojo::JSON::to_json($decoded_ltp_log);
+    Mojo::File::path($ltp_log)->spurt($final_ltp_log);
+}
+
+sub gen_ltp_env {
+    my $environment = {
+        product     => get_required_var('DISTRI') . ':' . get_required_var('VERSION'),
```

Comment:
BTW whole `$environment` section is nearly the same with `schedule_tests` in `lib/LTP/utils.pm`, you just use `PUBLIC_CLOUD_ARCH` and run `ver_linux` before. But understand that this is probably not worth of reusing in separate function. But maybe pass `ver_linux` output to `gen_ltp_env` to make sure it's filled (to prevent regressions during refactoring code flow in the future)?